tasks/memcached_exporter.yml

```diff
@@ -0,0 +1,45 @@
+- name: Download memcached_exporter tarball
+  get_url:
+    url: '{{ memcached_exporter_url }}'
+    dest: '/tmp/{{ memcached_exporter_release_name }}.tar.gz'
+    checksum: 'sha256:{{ memcached_exporter_checksum }}'
+    mode: 0644
+
+- name: Unpack tarball
+  unarchive:
+    src: '/tmp/{{ memcached_exporter_release_name }}.tar.gz'
+    dest: '/tmp'
+    creates: '/tmp/{{ memcached_exporter_release_name }}/memcached_exporter'
+    remote_src: true
+
+- name: Copy binary
+  copy:
+    src: '/tmp/{{ memcached_exporter_release_name }}/memcached_exporter'
+    dest: '/usr/local/bin/memcached_exporter'
+    remote_src: true
+    mode: 0755
+  notify: restart memcached_exporter
```

Comment:
You can avoid having to use the `copy` task by using `extra_ops` with `unarchive`:
```yaml
- name: Unpack binary from tarball
  unarchive:
    src: '/tmp/{{ memcached_exporter_release_name }}.tar.gz'
    dest: '/usr/local/bin'
    creates: '/usr/local/bin/memcached_exporter'
    remote_src: true
    extra_ops:
      - '--wildcards'
      - '*/memcached_exporter'
```

Reply:
works great, thanks